src/core/merge_right.rs

```diff
@@ -1,12 +1,19 @@
 use std::collections::{BTreeMap, BTreeSet, HashMap, HashSet};
-use std::sync::Arc;
 
-use serde_yaml::Value;
+use prost_reflect::prost_types::FileDescriptorProto;
+
+use super::valid::Valid;
 
 pub trait MergeRight {
     fn merge_right(self, other: Self) -> Self;
 }
 
+pub trait MergeRightNew {
```

Comment:
Do we want to go ahead with this name? Do we need two implementations?

Reply:
Can't find any use of this trait anywhere in the codebase either.

Reply:
ah, just some testing code left, I removed it